src/integrationTest/java/integration/tests/IsReadableTest.java

```diff
@@ -0,0 +1,68 @@
+package integration.tests;
+
+import com.firebolt.jdbc.connection.FireboltConnection;
+import integration.IntegrationTest;
+import lombok.CustomLog;
+import org.junit.jupiter.api.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static integration.tests.SystemEngineTest.SYSTEM_ENGINE_NAME;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@CustomLog
+public class IsReadableTest extends IntegrationTest {
+    @Test
+    void connection() throws SQLException {
+        try (Connection connection = createConnection()) {
+            log.info("connection test");
+            String database = ((FireboltConnection)connection).getSessionProperties().getDatabase();
+            log.info("database={}", database);
+            PreparedStatement ps1 = connection.prepareStatement("select engine_name from information_schema.engines where status = 'Running' and engine_type = 'General Purpose' and lower(attached_to) = lower(?)");
+            ps1.setString(1, database);
+            ResultSet rs1 = ps1.executeQuery();
+            while (rs1.next()) {
+                String engineName = rs1.getString(1);
+                log.info("engineName={}", engineName);
+            }
+            log.info("end of the engines list");
+
+            PreparedStatement ps2 = connection.prepareStatement("select count(*) = 0 from information_schema.engines where status = 'Running' and engine_type = 'General Purpose' and lower(attached_to) = lower(?)");
+            ps2.setString(1, database);
+            ResultSet rs2 = ps2.executeQuery();
+            while (rs2.next()) {
+                boolean isReadOnly = rs2.getBoolean(1);
+                log.info("read only={}", isReadOnly);
+            }
+            log.info("end of the readonly");
```

Comment:
Why are we doing those statements in a test? Is this leftover from debugging?

Reply:
no, this code is added for debugging.

Reply:
removed redundant code